device/Microsoft.Azure.Devices.Client/TransientFaultHandling/AsyncExecution.cs

```diff
@@ -30,11 +30,11 @@ namespace Microsoft.Azure.Devices.Client.TransientFaultHandling
     /// Provides a wrapper for a non-generic <see cref="T:System.Threading.Tasks.Task" /> and calls into the pipeline
     /// to retry only the generic version of the <see cref="T:System.Threading.Tasks.Task" />.
     /// </summary>
-    internal class AsyncExecution : AsyncExecution<bool>
+    internal class AsyncExecution : AsyncExecution<Task<bool>>
```

Comment:
I don't understand why the change is within the Microsoft Patterns & Practices code. 
If we have a leak caused by this, we are most likely using it incorrectly, not that the definition is incorrect.